brave-spark-framework-filter/src/main/java/com/github/kristofa/brave/spark/internal/MaybeAddClientAddressFromRequest.java

```diff
@@ -0,0 +1,36 @@
+package com.github.kristofa.brave.spark.internal;
+
+import com.github.kristofa.brave.Brave;
+import com.github.kristofa.brave.ServerResponseInterceptor;
+import com.github.kristofa.brave.internal.MaybeAddClientAddress;
+import zipkin.Constants;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * Parses the {@link Constants#CLIENT_ADDR client address}, by looking at "X-Forwarded-For", then
+ * the remote address of the request. This performs no DNS lookups.
+ *
+ * <p>This is a hack as {@link ServerResponseInterceptor} doesn't yet support client addresses.
+ */
+public final class MaybeAddClientAddressFromRequest
```

Comment:
add a dependency on web-servlet-filter so you don't have to copy/paste this and also its tests